src/Decoder.c

```diff
@@ -2313,13 +2313,25 @@ static void ZydisSetAttributes(ZydisDecoderContext* context, ZydisDecodedInstruc
             default:
                 break;
             }
-        } else
+        }
+
+        if (def->accepts_NOTRACK)
         {
-            if (def->accepts_segment)
+            instruction->attributes |= ZYDIS_ATTRIB_ACCEPTS_NOTRACK;
+            if (context->decoder->decoder_mode[ZYDIS_DECODER_MODE_CET] &&
+                (context->prefixes.effective_segment == 0x3E))
             {
-                instruction->attributes |= ZYDIS_ATTRIB_ACCEPTS_SEGMENT;
+                instruction->attributes |= ZYDIS_ATTRIB_HAS_NOTRACK;
+                instruction->raw.prefixes[context->prefixes.offset_group2].type =
+                    ZYDIS_PREFIX_TYPE_EFFECTIVE;
             }
-            if (context->prefixes.effective_segment && def->accepts_segment)
+        }
+
+        if (def->accepts_segment && !def->accepts_branch_hints)
+        {
+            instruction->attributes |= ZYDIS_ATTRIB_ACCEPTS_SEGMENT;
+            if (context->prefixes.effective_segment && 
+                !(instruction->attributes & ZYDIS_ATTRIB_HAS_NOTRACK))
```

Comment:
Regarding @tremalrik comment:

> Intel's documentation doesn't make it clear whether the notrack prefix still retains its old segment-override function when CET is enabled in legacy/compatibility mode. A couple of examples of 32-bit instruction encodings where that distinction would matter are:
>
> - 3E FF 55 00 : is "notrack call ds:[ebp]" if it is recognized as a segment prefix or "notrack call ss:[ebp]" if it isn't.
> - 26 3E FF 10 : is "notrack call ds:[eax]" if it is recognized as a segment prefix or "notrack call es:[eax]" if it isn't.

For me it makes most sense, if the default group 2 behavior applies to the `no-track` prefix as well: Only last group 2 prefix counts, rest is ignored (special case in 64-bit mode, where `0x64`/`0x65` always has precedence). I did not test this on actual hardware and did not find any information either tho.

Zydis will output:
- `3E FF 55 00` : `notrack call dword ptr ss:[ebp]`
- `26 3E FF 10` : `notrack call dword ptr ds:[eax]`